src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/controllers/TransformationControllerTest.java

```diff
@@ -0,0 +1,165 @@
+package uk.gov.hmcts.reform.bulkscanccdeventhandler.transformation.controllers;
+
+import com.microsoft.applicationinsights.web.dependencies.apachecommons.lang3.tuple.Pair;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+import uk.gov.hmcts.reform.authorisation.exceptions.InvalidTokenException;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.common.auth.AuthService;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.common.auth.ForbiddenException;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.common.auth.UnauthenticatedException;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.transformation.model.out.Address;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.transformation.model.out.CaseCreationDetails;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.transformation.model.out.SampleCase;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.transformation.model.out.ScannedDocument;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.transformation.model.out.SuccessfulTransformationResponse;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.transformation.services.ExceptionRecordToCaseTransformer;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import static java.time.format.DateTimeFormatter.ISO_DATE_TIME;
+import static java.util.Arrays.asList;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.springframework.http.HttpStatus.FORBIDDEN;
+import static org.springframework.http.HttpStatus.UNAUTHORIZED;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(TransformationController.class)
+public class TransformationControllerTest {
+
+    @Autowired private MockMvc mockMvc;
+
+    @MockBean private ExceptionRecordToCaseTransformer transformer;
+    @MockBean private AuthService authService;
+
+    @Test
+    public void should_return_proper_status_codes_for_auth_exceptions() throws Exception {
+        List<Pair<RuntimeException, HttpStatus>> exceptionsAndStatuses =
+            asList(
+                Pair.of(new UnauthenticatedException(null), UNAUTHORIZED),
+                Pair.of(new InvalidTokenException(null, null), UNAUTHORIZED),
+                Pair.of(new ForbiddenException(null), FORBIDDEN)
+            );
+
+        for (Pair<RuntimeException, HttpStatus> pair : exceptionsAndStatuses) {
```

Comment:
mockMvc throws checked exceptions, can call `.foreach(x -> ...)` on list above 😢

Reply:
How about extracting loop source into parameterised test source?

```java
private static Stream<Arguments> exceptionsAndStatuses() {
    return Stream.of(
      Arguments.of(new UnauthenticatedException(null), UNAUTHORIZED),
      Arguments.of(new InvalidTokenException(null, null), UNAUTHORIZED),
      Arguments.of(new ForbiddenException(null), FORBIDDEN)
    );
}

@ParameterizedTest
@MethodSource("exceptionsAndStatuses")
public void should_return_proper_status_codes_for_auth_exceptions(
    RuntimeException exception,
    HttpStatus expectedStatus
) throws Exception {
    reset(authService);
    given(authService.authenticate(any())).willThrow(exception);

    sendRequest("{}").andExpect(status().is(expectedStatus.value()));
}
```

can easily extend the source

Reply:
done